src/AlignmentGuides.js

```diff
@@ -152,6 +177,47 @@ class AlignmentGuides extends Component {
 			}
 		}
 
+		if (this.props?.groups !== prevProps.groups) {
+			let {boxes} = this.state;
+			boxes = Object.fromEntries(
+				Object.entries(boxes)
+				  .filter(([key]) => !key.startsWith('box-ms-'))
+			  )
+			if (this.props?.groups?.length === 0) {
+				this.setState({
+					boxes
+				});
+			} else if (this.props?.groups?.length > 0) {
+				let active = this.state.active;
+				this.props?.groups?.forEach((groupArray, index) => {
```

Comment:
This logic is also written inside `componentDidMount`. Can we extract common parts and create a utility function?